src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs

```diff
@@ -17,12 +18,16 @@ namespace Microsoft.PowerFx.Functions
 {
     internal static partial class Library
     {
-        public delegate FormulaValue FunctionPtr(EvalVisitor runner, SymbolContext symbolContext, IRContext irContext, FormulaValue[] args);
+        // Sync FunctionPtr - all args are evaluated before invoking this function.  
+        public delegate FormulaValue FunctionPtr(SymbolContext symbolContext, IRContext irContext, FormulaValue[] args);
```

Comment:
>SymbolContext [](http://example.com/codeflow?start=49&length=13)

If a function invokes lambdas, then it's could be async, and so needs to be a AsyncFunctionPtr. 
If it doesn't invoke lambas, then it doesn't need a EvalVisitor parameter.

Reply:
This causes some pain in the standard error handling wrappers, because the library functions sometimes omit 2/4 parameters.